src/verilog/verilog_typecheck_expr.cpp

```diff
@@ -1344,7 +1349,16 @@ exprt verilog_typecheck_exprt::elaborate_constant_expression(exprt expr)
 
     // We fall back to the simplifier to approximate
     // the standard's definition of 'constant expression'.
-    return simplify_expr(expr, ns);
+    auto simplified_expr = simplify_expr(expr, ns);
```

Comment:
Because we've been bitten by this in CBMC (cf. https://github.com/diffblue/cbmc/pull/7856): any use of the simplifier in the front-end needs to make sure that the resulting expression still has a source location.

Reply:
added